src/zenml/zen_server/deploy/helm/templates/server-db-job.yaml

```diff
@@ -0,0 +1,141 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "zenml.fullname" . }}-db-migration
+  labels:
+    {{- include "zenml.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": pre-install,pre-upgrade
+    "helm.sh/hook-weight": "-1"
+    "helm.sh/hook-delete-policy": hook-succeeded
+{{- if .Values.zenml.database.url }}
```

Comment:
I think this condition should guard the entire job definition instead of just its `spec`.